github_issue_summarization/pipelines/example_pipelines/pipelines-notebook.ipynb

```diff
@@ -361,47 +357,42 @@
     "    )\n",
     "\n",
     "  train = train_op(\n",
-    "    working_dir=working_dir,\n",
     "    data_dir=data_dir,\n",
-    "    checkpoint_dir=checkpoint_dir,\n",
-    "    model_dir='%s/%s/model_output' % (working_dir, '{{workflow.name}}'),\n",
+    "    model_dir=copydata.outputs['copy_output_path'],\n",
     "    action=TRAIN_ACTION, train_steps=train_steps,\n",
     "    deploy_webapp=deploy_webapp\n",
     "    ).apply(gcp.use_gcp_secret('user-gcp-sa'))\n",
     "\n",
     "  log_dataset.after(copydata)\n",
-    "  train.after(copydata)\n",
     "  train.after(preproc)\n",
     "\n",
     "  log_model = metadata_log_op(\n",
     "    log_type=MODEL,\n",
     "    workspace_name=WORKSPACE_NAME,\n",
     "    run_name='{{workflow.name}}',\n",
-    "    model_uri='%s/%s/model_output' % (working_dir, '{{workflow.name}}')\n",
+    "    model_uri=copydata.outputs['copy_output_path']\n",
     "    )\n",
     "\n",
     "  serve = dsl.ContainerOp(\n",
     "      name='serve',\n",
     "      image='gcr.io/google-samples/ml-pipeline-kubeflow-tfserve',\n",
     "      arguments=[\"--model_name\", 'ghsumm-%s' % ('{{workflow.name}}',),\n",
-    "          \"--model_path\", '%s/%s/model_output/export' % (working_dir, '{{workflow.name}}')\n",
+    "          \"--model_path\", train.outputs['train_output_path']\n",
     "          ]\n",
     "      )\n",
-    "\n",
     "  log_model.after(train)\n",
-    "  serve.after(train)\n",
-    "  train.set_gpu_limit(4)\n",
+    "  train.set_gpu_limit(1)\n",
     "  train.set_memory_limit('48G')\n",
     "\n",
-    "  with dsl.Condition(train.output == 'true'):\n",
+    "  with dsl.Condition(train.outputs['launch_server'] == 'true'):\n",
     "    webapp = dsl.ContainerOp(\n",
     "        name='webapp',\n",
     "        image='gcr.io/google-samples/ml-pipeline-webapp-launcher:v2ap',\n",
     "        arguments=[\"--model_name\", 'ghsumm-%s' % ('{{workflow.name}}',),\n",
```

Comment:
'{{workflow.name}}' should ideally be replaced with `kfp.dsl.RUN_ID_PLACEHOLDER` (so that we do not expose Argo). (`RUN_ID_PLACEHOLDER` == `{{workflow.uid}}`, not `{{workflow.name}}`, but there should not be a big difference).

Reply:
I went ahead and made this change as well.